app/services/org_nummers_generator.rb

```diff
@@ -0,0 +1,52 @@
+# Generate a valid Swedish OrganisationNummer (OrgNummer; OrgNr).
+#
+# It must be 10 digits and the last digit must be the valid Luhn checksum, per
+# https://sv.wikipedia.org/wiki/Organisationsnummer
+#
+
+require 'orgnummer'
+require 'set'
+
+class OrgNummersGenerator
+
+  MAX_NUM = 9999999999 unless defined?(MAX_NUM)
+  MIN_NUM = 1000000000 unless defined?(MIN_NUM)  # orgnummer gem will not recognize any number less than this as valid
+
+  NONE_FOUND = nil  unless defined?(NONE_FOUND)
+
+
+  def self.generate(number_to_generate = 1)
+
+    results = Set.new
+
+    num_tries = 0
+    all_possible = MAX_NUM - MIN_NUM
+
+    # num_tries keeps use from trying forever.
+    # It is a somewhat arbitrary limit on how many times we can try.
+    # Even though each try is a Random number, so it is possible to get the same number more than once,
+    # this is a reasonable (if clunky) limit.
+    while results.count < number_to_generate && (num_tries < all_possible) do
+      number_to_generate.times do
+        org_num = generate_one
+        num_tries += 1 if results.add?(org_num)
+      end
+
+    end
```

Comment:
When I read this method it appears to say that we will try _all possible_ numbers before giving up.  

But really what is happening is that we're selecting a random starting number (_all possible_ times), and then incrementing that number until we find a valid number.

So, consider just using a _max_tries_ constant instead. (the similar `seeds.rb` method uses 100).